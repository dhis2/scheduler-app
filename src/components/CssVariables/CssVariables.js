import { theme, colors } from '@dhis2/ui-core'

const root = document.documentElement

const CssVariables = () => {
    Object.keys(theme).forEach(key => {
        root.style.setProperty(`--theme-${key}`, theme[key])
    })

    Object.keys(colors).forEach(key => {
        root.style.setProperty(`--colors-${key}`, colors[key])
    })

    return null
}

export default CssVariables
