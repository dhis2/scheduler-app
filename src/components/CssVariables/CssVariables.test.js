import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { theme, colors } from '@dhis2/ui-core'
import CssVariables from './CssVariables'

describe('<CssVariables>', () => {
    it('inserts css theme variables', () => {
        const div = document.createElement('div')
        const root = document.documentElement

        ReactDOM.render(<CssVariables />, div)

        Object.keys(theme).forEach(key => {
            const value = getComputedStyle(root).getPropertyValue(
                `--theme-${key}`
            )
            expect(value).toEqual(theme[key])
        })

        ReactDOM.unmountComponentAtNode(div)
    })

    it('inserts css color variables', () => {
        const div = document.createElement('div')
        const root = document.documentElement

        ReactDOM.render(<CssVariables />, div)

        Object.keys(colors).forEach(key => {
            const value = getComputedStyle(root).getPropertyValue(
                `--colors-${key}`
            )
            expect(value).toEqual(colors[key])
        })

        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders correctly', () => {
        const wrapper = shallow(<CssVariables />)

        expect(wrapper).toMatchSnapshot()
    })
})
