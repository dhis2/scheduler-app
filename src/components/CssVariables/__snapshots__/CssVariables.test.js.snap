// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CssVariables> renders correctly 1`] = `""`;
